datafusion/optimizer/src/push_down_filter.rs

```diff
@@ -441,11 +442,11 @@ fn push_down_all_join(
 
     // Extract from OR clause, generate new predicates for both side of join if possible.
     // We only track the unpushable predicates above.
-    if left_preserved {
+    if on_left_preserved {
```

Comment:
`keep_predicates`, `join_conditions` are join filter but not predicates from top Filter, we should use `on_lr_is_preserved` instead of `lr_is_preserved`.